apollo-sample/src/main/java/com/example/apollostack/sample/TestQueryWithFragmentResponse.java

```diff
@@ -0,0 +1,198 @@
+          }
+        });
+      } else {
+        streamReader.skipNext();
+      }
+    }
+  }
+
+  @Nonnull public Data getData() {
+    return data;
+  }
+
+  public static class Data implements Operation.Data {
+    private @Nullable AllPeople allPeople;
+
+    public Data(ResponseStreamReader streamReader) throws IOException {
+      while (streamReader.hasNext()) {
+        String nextName = streamReader.nextName();
+        if ("allPeople".equals(nextName)) {
+          this.allPeople = streamReader.nextOptionalObject(new ResponseStreamReader.NestedReader<AllPeople>() {
+            @Override public AllPeople read(ResponseStreamReader reader) throws IOException {
+              return new AllPeople(reader);
+            }
+          });
+        } else {
+          streamReader.skipNext();
+        }
+      }
+    }
+
+    @Nullable public AllPeople getAllPeople() {
+      return allPeople;
+    }
+
+    public @Nullable AllPeople allPeople() {
+      return this.allPeople;
+    }
+
+    public static class AllPeople {
+      private @Nullable Integer totalCount;
+      private @Nullable List<? extends Edge> edges;
+
+      public AllPeople(ResponseStreamReader streamReader) throws IOException {
+        while (streamReader.hasNext()) {
+          String nextName = streamReader.nextName();
+          if ("totalCount".equals(nextName)) {
+            this.totalCount = streamReader.nextOptionalInt();
+          } else if ("edges".equals(nextName)) {
+            this.edges = streamReader.nextList(new ResponseStreamReader.NestedReader<Edge>() {
+              @Override public Edge read(ResponseStreamReader reader) throws IOException {
+                return reader.nextObject(new ResponseStreamReader.NestedReader<Edge>() {
+                  @Override public Edge read(ResponseStreamReader reader) throws IOException {
+                    return new Edge(reader);
+                  }
+                });
+              }
+            });
+          } else {
+            streamReader.skipNext();
+          }
+        }
+      }
+
+      public @Nullable Integer totalCount() {
+        return this.totalCount;
+      }
+
+      public @Nullable List<? extends Edge> edges() {
+        return this.edges;
+      }
+
+      public static class Edge {
+        private @Nonnull String cursor;
+        private @Nullable Node node;
+
+        public Edge(ResponseStreamReader streamReader) throws IOException {
+          while (streamReader.hasNext()) {
+            String nextName = streamReader.nextName();
+            if ("cursor".equals(nextName)) {
+              this.cursor = streamReader.nextString();
+            } else if ("node".equals(nextName)) {
+              this.node = streamReader.nextOptionalObject(new ResponseStreamReader.NestedReader<Node>() {
+                @Override public Node read(ResponseStreamReader reader) throws IOException {
+                  return new Node(new BufferedResponseStreamReader(reader));
```

Comment:
For objects that define fragments we switch from stream reader to buffered reader